op-chain-ops/genesis/layer_two_test.go

```diff
@@ -94,3 +93,110 @@ func TestBuildL2DeveloperGenesisDevAccountsFunding(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 2321, len(gen.Alloc))
 }
+
+func TestBuildL2MainnetGenesis(t *testing.T) {
+	config, err := genesis.NewDeployConfig("./testdata/test-deploy-config-devnet-l1.json")
+	require.Nil(t, err)
+
+	backend := backends.NewSimulatedBackend(
+		core.GenesisAlloc{
+			crypto.PubkeyToAddress(testKey.PublicKey): {Balance: big.NewInt(10000000000000000)},
+		},
+		15000000,
+	)
+	block, err := backend.BlockByNumber(context.Background(), common.Big0)
+	require.NoError(t, err)
+
+	gen, err := genesis.BuildL2MainnetGenesis(config, block)
+	require.Nil(t, err)
+	require.NotNil(t, gen)
+
+	depB, err := bindings.GetDeployedBytecode("Proxy")
+	require.NoError(t, err)
+
+	for name, predeploy := range predeploys.Predeploys {
+		addr := *predeploy
+		if predeploys.IsDeprecated(addr) {
+			continue
+		}
+
+		account, ok := gen.Alloc[addr]
+
+		if predeploys.IsDeprecated(addr) && !predeploys.IsProxied(addr) {
+			require.Equal(t, false, ok, name)
+			continue
+		}
+		require.Equal(t, true, ok, name)
+		require.Greater(t, len(account.Code), 0)
+
+		if !predeploys.IsProxied(addr) {
+			continue
+		}
+
+		adminSlot, ok := account.Storage[genesis.AdminSlot]
+		require.Equal(t, true, ok)
+		require.Equal(t, predeploys.ProxyAdminAddr.Hash(), adminSlot)
+		require.Equal(t, depB, account.Code)
+	}
+	require.Equal(t, 2063, len(gen.Alloc)) // TODO: confirm 2063 is correct!
```

Comment:
Do you think a manual inspection of the output would be appropriate to accomplish this todo?